### DISCUSSION TITLE
I need to know which OS type to use for rclone to work on my synology-nas

### DISCUSSION THREAD

**Discussion by NovaZazen** [itsToggle/plex_debrid#332] [Q&A]
I'm wanting to set all this up on my synology-nas but it seems they have some kind of proprietary OS that is working off a linux based OS. I'm about to download rclone from https://rclone.org/downloads/ and there are many options for architecture and OS. My synology seems to be an AMD 64-bit system, so I at least know the architecture. Between these options "Linux", "FreeBSD", "NetBSD", "OpenBSD", "Solaris" I need to know which will work with my system...if any. Someone told me that any of these options should work seeing they are all linux based OS's but I don't want to go on a whim and mess up my NAS in the process. Any help with this would be greatly appreciated.

**Comment by itsToggle** [comment]
you can simply try them out. You cant mess up your system trying to run programs for a different OS / architecture, they will simply not start :)
If the version you tried out wasnt ment for your NAS, you will most probable get a "Syntax Error". There is another type of error related to this, but I cant remember what the error message for that would be.
Anyways, just try them out if you arent sure or consult google about your NAS OS and architecture.

**Reply by NovaZazen** [reply]
Thanks for replying. I couldn't get any of the OS versions to work...every file type says its an invalid package type. I am trying to launch rclone inside a docker container on my nas now, but every time I run it it says it has stopped unexpectedly and to look at the docker logs. I go to the logs and there isn't anything there. I think there is something going wrong with how I am mounting the rclone container to my NAS. I am wanting to create the virtual drive for debrid that communicates to one of my shared folders, as I need this in order to access the folder via my plex server. I've been googling like a mad man, and I can't come up on anything. All I have found are command line suggestions to be run via putty through ssh to my nas. I did this but the commands come back with permission denied even though I have logged in as an admin. I am kind of losing my mind here, any help would be greatly appreciated. I posted some screen shots in order of what I am doing to set this up through docker.


![Screenshot (9)](https://user-images.githubusercontent.com/112219603/213153495-d3271852-4328-4c59-961a-7388a3f4269a.png)
![Screenshot (10)](https://user-images.githubusercontent.com/112219603/213153504-9db37fbb-c2ee-4a75-99a2-fc2c09161c07.png)
![Screenshot (11)](https://user-images.githubusercontent.com/112219603/213153516-7ac6699d-5553-4624-bdeb-e9b76ad97adb.png)
![Screenshot (12)](https://user-images.githubusercontent.com/112219603/213153526-a4244504-f224-4c1f-bb2a-7715bf26a152.png)
![Screenshot (13)](https://user-images.githubusercontent.com/112219603/213153535-3f2afe8e-9e77-4bf5-9397-5ab4bf6f8e86.png)
![Screenshot (14)](https://user-images.githubusercontent.com/112219603/213153547-79fc7be1-fb34-43d0-a74b-1ab586770f5a.png)
![Screenshot (15)](https://user-images.githubusercontent.com/112219603/213153554-6c59a491-fe35-4fcb-a8e3-8a1398640f3e.png)
![Screenshot (16)](https://user-images.githubusercontent.com/112219603/213153562-6c00ac41-cf34-4473-95ca-1433eb6e61c6.png)
![Screenshot (17)](https://user-images.githubusercontent.com/112219603/213153573-9d9945af-b086-4c2b-bfc7-e63b0e2072f0.png)
![Screenshot (18)](https://user-images.githubusercontent.com/112219603/213153584-634b25c8-218a-40fe-afee-2e59319c919c.png)

**Reply by itsToggle** [reply]
permission denied might simply be that the rclone file is not marked as an excecutable. Have you attempted to run `chmod u+x your-filename` ?

**Reply by NovaZazen** [reply]
I have to no avail. I really would rather avoid going to the terminal as my NAS doesn't really like cooperating in that department. The rclone image is supported via the docker package on my NAS. I'm having issues getting it to run however as I made mention of earlier. I put a series of screenshots showing the steps I'm taking to set up the container in docker. Is there anything that I am doing wrong? The one thing that I not sure about is the mount path. I'm not sure where to mount the container to and I believe that may be causing issues. i've tried all sorts of different paths and I'm not sure exactly where I am supposed to put it.

**Reply by NovaZazen** [reply]
Would you or anyone else be willing to get on a call to try and sort this out? I'm really running into a brick wall it feels like and any advice would be immensely helpful.

**Reply by itsToggle** [reply]
Hoe about you hop on our discord server? I'm sure the people on there will happily help you out.

**Comment by jamesah2008** [comment]
NovaZazen, I'm also running up against the same challenge. I'd be happy to hop on and troubleshoot with you.

**Reply by NovaZazen** [reply]
That would be great. Would you be available tomorrow? I've been making some headway but am still yet to solve the issues I've been having. As old issues are solved new ones arrive 😂.

My discord is NovaZazen#3424 if you want to add me there.
